ui/lib/core/addon/components/secret-list-header.js

```diff
@@ -48,7 +48,7 @@ export default class SecretListHeader extends Component {
 
   @action
   transitionToNewPki() {
-    this.router.transitionTo('vault.cluster.secrets.backend.pki.overview', this.args.model.engineType);
```

Comment:
I think `engineType` would always resolve to `pki` and the dynamic segment needs to be the id of the `secret-engine` so `this.args.model.id` should work. I assume that the transition happens between child routes of `vault.cluster.secrets.backend` so the model would already be loaded but more of an FYI.

Reply:
@zofskeez I wrote this in the ticket I filed - want to make sure I understand ember routing since it often bonks me up 😅. I suggested removing because of where this transition is happening, but want to make sure I'm thinking about it correctly 

> It looks like in the transitionToNewPki method in SecretListHeader we can remove the second arg: this.args.model.engineType since ember sets the backend part of the dynamic route as the mount currently being viewed. (which we’re transitioning from)

Reply:
@hellobontempo ya that's how I'm understanding it. The from and to routes involved in the transition have the same parent which loads the `secret-engine` model so it is able to resolve the dynamic segment. It's probably best to always include the value for the dynamic segment but in this case it is the only place the transition can be triggered so the parent route should always have the model.

Reply:
Good to know! Thank you!